Add integration tests example and usage within emulator (#1)
Build-Squad/flow-code-coverage: README.md

- PASS: testMap
- PASS: testMapStrings
- PASS: testReduce
Coverage: 90.6% of statements
```

> Look at the files `ArrayUtils.cdc` (smart contract) and `test_array_utils.cdc` (tests for the smart contract). For the `ArrayUtils.range` method, we have omitted in purpose the branch where `start > end`. It is left as an exercise to the reader. Look at the comment on line 25 in `test_array_utils.cdc`.

Reviewer: ```suggestion
Look at the files `ArrayUtils.cdc` (smart contract) and `test_array_utils.cdc` (tests for the smart contract). 
For the `ArrayUtils.range` method, we have omitted the code branch where `start > end` on purpose. It is left as an exercise for the reader. Look at the comment on line 25 in `test_array_utils.cdc`.
```

Author: Updated in https://github.com/Build-Squad/flow-code-coverage/pull/1/commits/4fd71019aea467b88db1ed29592dc79b2eca7176

- PASS: testSubstringUntil
- PASS: testSplit
- PASS: testJoin
Coverage: 55.5% of statements
```

The generated `coverage.json` file is somewhat more elaborate, for integration tests. By viewing its content, we find the following keys:

- `A.01cf0e2f2f715450.ArrayUtils`
- `A.01cf0e2f2f715450.StringUtils`
- `A.0ae53cb6e3f42a79.FlowToken`
- `A.e5a8b7f23e8b548f.FlowFees`
- `A.ee82856bf20e2aa6.FungibleToken`
- `A.f8d6e0586b0a20c7.FlowServiceAccount`

> and some other locations. The ones starting with `s.*` are scripts, while the ones starting with `t.*` are transactions.

Reviewer: ```suggestion
and some other locations.

Locations that start with `A.` are contracts deployed to an account, ones that start with `s.` are scripts, and ones that start with `t.` are transactions.
```

Author: Updated in https://github.com/Build-Squad/flow-code-coverage/pull/1/commits/d61f8dbce7558878cf4b518c144045c7d91f292b

The rest of the keys are system contracts that are always available in the Flow Emulator, which is utilized as the backend implementation for integration tests.

## For Emulator

It is also possible to view code coverage through the emulator, outside the context of testing.

All we have to do is start the emulator with the necessary flag (`coverage-reporting`):

```bash
flow emulator --storage-limit=false --coverage-reporting -v
```

With this, we can use our browser and visit http://localhost:8080/emulator/codeCoverage.

This code coverage report will reflect every interaction with the emulator. For example, we can deploy contracts to the emulator, run scripts/transactions against them and view the results:

```bash
flow deploy contracts --network=emulator

flow scripts execute foo_contract_scripts.cdc --network=emulator
```

![Emulator Code Coverage](./emulator-code-coverage.png)

We can also flush/reset the collected code coverage report, with:

```bash
curl -XPUT 'http://localhost:8080/emulator/codeCoverage/reset'
```

Which results in the following:

![Code Coverage Reset](./code-coverage-reset.png)

> All of the keys have disappeared, except for the `A.f8d6e0586b0a20c7.FlowServiceAccount`, which is a system contract that runs frequently some operations.

Reviewer: ```suggestion
All of the keys have disappeared, except for `A.f8d6e0586b0a20c7.FlowServiceAccount`, which is a system contract that is essential to the operations of Flow.
```

Author: Updated in https://github.com/Build-Squad/flow-code-coverage/pull/1/commits/385067c991608e7a37eba742bd857fab136ebeb0
